gv.c

```diff
@@ -4018,6 +4018,15 @@ Perl_amagic_call(pTHX_ SV *left, SV *right, int method, int flags)
     }
   }
 
+  /* If there's an optimised-away assignment such as $lex = $a + $b, where
+   * the  operator sets the targ lexical directly and skips the sassign,
+   * treat the op as scalar even if its marked as void */
+  if (   PL_op
+      && (PL_opargs[PL_op->op_type] & OA_TARGLEX)
+      && (PL_op->op_private & OPpTARGET_MY)
+  )
+      force_scalar = 1;
```

Comment:
Combined with:
```
    U8 gimme = (force_scalar || PL_op->op_type == OP_MULTICONCAT)
                    ? G_SCALAR : GIMME_V;
```
also in this function, Coverity scan complains that we check whether `PL_op` is non-NULL in this newly added code, but don't check in the assignment to `gimme`.

Is it valid for `PL_op` to be NULL for a call to Perl_amagic_call()?